### PR TITLE
Budget-app V 1.0

### PR DESCRIPTION
**Pull request by BregornOriginal**
# This is the first version of this Budget-App 💯 

## - Splash screen 👍🏼 

    - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
 
## - Sign up and log in pages 👍🏼

    - The user should be able to register in the app with full name, email and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

## - Home page (catagories page) 👍🏼

    - When the user logs in, they are presented with the categories page.
    - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

## - Transactions page 👍🏼

    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

## - "Add a new category" page 👍🏼

    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.
 
## - "Add a new transaction" page 👍🏼

    - The user fills out a form to create a new transaction with:
        - name (mandatory)
        - amount (mandatory)
        - categories (mandatory at least one)
    - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

## - Testing requirements 👍🏼
    - Create unit and integration tests for all the most important components of your RoR application.

- **Used Postgres as your database**.
- **Used devise for authentication**.
- **Validated all user input to make sure that anyone with bad intentions cannot compromise your app**.

## Presentation of the APP in a LOOM VIDEO

[Library | Loom - 13 October 2022 - Watch Video](https://www.loom.com/share/f62dbfc6d8404050ab084fd09fb70b81)

## Live version on Heroku

[Budget-App](https://bregorn-budget-app.herokuapp.com/)

## Example of the app

blog-app-view

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+## Pesentation
+
+https://www.loom.com/share/1492d3e6e83e40bcaa6131f4c9410839
 
 ## Live Version
 
-[Avaible Soon]()
+[Avaible Soon on Heroku]()
+
+## Demo
```
Comment (arslanbisharat) [inline README.md:27]: - [ ] Please make sure the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: README.md [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+
+![blog-app-view](https://user-images.githubusercontent.com/93630700/195625663-549514a9-6b19-4387-b897-5ebbd08bf21f.gif)
```
Comment (arslanbisharat) [inline README.md:29]: - [ ] In the recording, please share which part of your application or code you liked the most and why,

File: README.md [hunk 3]
```diff
@@ -1,9 +1,33 @@
...
+The style followed is a mobile-first structure.
+
+## Pesentation
+
+https://www.loom.com/share/1492d3e6e83e40bcaa6131f4c9410839
 
 ## Live Version
 
-[Avaible Soon]()
+[Avaible Soon on Heroku]()
```
Comment (arslanbisharat) [inline README.md:25]: - [ ] Please provide us with a proper link to your application. This is one of the basic requirements for approval of this project.

File: app/controllers/user_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_user
+    @user = User.find(params[:id])
```
Comment (arslanbisharat) [inline app/controllers/user_controller.rb:29]: - [ ] [OPTIONAL] For this project you have used the correct coloring scheme and typography but the decision doesn't 100% meet with the design provided to you. Such as a complete page, add new categories page, etc. I recommend you follow the design as much as possible.

File: config/environments/test.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+require "active_support/core_ext/integer/time"
+
```
Comment (arslanbisharat) [inline config/environments/test.rb:2]: - [ ] [OPTIONA] It would be great if you can add more tests.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @BregornOriginal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (BregornOriginal) [reply README.md:27]: Hi @arslanbisharat Thanks for your suggestions, I totally forgot to mention Gregoire Vella. Now it's done!

File: README.md (hunk 2 above)
Comment (BregornOriginal) [reply README.md:29]: I've recorded a new video, please check if that's ok now!

[Library | Loom - 13 October 2022 - Watch Video](https://www.loom.com/share/f62dbfc6d8404050ab084fd09fb70b81)

File: config/environments/test.rb (hunk 5 above)
Comment (BregornOriginal) [reply config/environments/test.rb:2]: Thanks for the suggestion! I'll try to improve it in the next few days!

File: README.md (hunk 3 above)
Comment (BregornOriginal) [reply README.md]: Done! https://bregorn-budget-app.herokuapp.com/

**Review by t-yanick — APPROVED** [review]
Hi @BregornOriginal, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Great job with test coverage
:heavy_check_mark: Great job with video presentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**